src/config.rs

```diff
@@ -110,6 +110,18 @@ pub enum Action {
     },
 }
 
+fn speed_default() -> Option<u64> {
```

Comment:
Could these default functions pertain to an `impl` block of the `enum Action`?

Reply:
You mean something like that?

```
...
        #[serde(skip_serializing_if = "Option::is_none", default = "Action::speed_default")]
        speed: Option<u64>,
...
impl Action {
    fn speed_default() -> Option<u64> {
        Some(50)
    }
}
...
```
Yes, it can, I can refactor all of them to be associated functions if you think it's better.

Reply:
I'm not sure. It is a static function anyway. So it does not matter. I thought of it for having the defaults contained where they will be used. But it is not that important. We can leave it like it is for now.

Reply:
No, I agree that it's better, will fix now :)

Reply:
Done.
Also did some changes to work with references instead of `clone()` where possible.